android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1096,7 +1154,30 @@ public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newStat
                                 linearLayoutManager.findFirstVisibleItemPosition();
                         int lastVisibleItemPosition =
                                 linearLayoutManager.findLastVisibleItemPosition();
-
+                        int scrollY = mParentScrollView.getScrollY();
+
+                        if (BraveActivity.getBraveActivity() != null
+                                && BraveActivity.getBraveActivity().getActivityTab() != null) {
+                            Tab tab = BraveActivity.getBraveActivity().getActivityTab();
+                            if (firstCompletelyVisibleItemPosition >= 0) {
+                                int position = 0;
+                                if (firstCompletelyVisibleItemPosition == 0
+                                        && ((ConfigurationUtils.isLandscape(mActivity)
+                                                    && scrollY > 600)
+                                                || (!ConfigurationUtils.isLandscape(mActivity)
+                                                        && scrollY > 900))) {
+                                    position = -2;
+                                } else {
+                                    position = firstCompletelyVisibleItemPosition == 0
+                                            ? -1
+                                            : firstCompletelyVisibleItemPosition;
+                                }
+                                if (tab != null) {
```

Comment:
this is always true, as you have `&& BraveActivity.getBraveActivity().getActivityTab() != null)` above

Reply:
refactored